### ISSUE TITLE
Cannot start container - docker_gwbridge: exchange full

### ISSUE THREAD

**Issue by sirlp2** [docker/docker#31100]
**Description**
Service Replica(container) fails to start with error:
```
starting container failed: container 27bd4b870733d866e62d3a7ba3367466984aaca442db8450a9746305762d9837: endpoint create on GW Network failed: failed to create endpoint gateway_27bd4b870733 on network docker_gwbridge: adding interface vethaa5c11e to bridge docker_gwbridge failed: exchange full
```

I have seen this same error in prior release issues.  What is the cause of this type of issue and what measures can be taken to prevent it?

**Steps to reproduce the issue:**
1. Created a new service with one replica using `--publish mode=host` for the port mapping
2. Increase the service replicas to 2
3. Watch the `docker_gwbridge` container list grow until it is exhausted
4. Attempt to start a new service and the error is encountered `docker_gwbridge failed: exchange full`

**Describe the results you received:**
Receive the above error trying to start a new service.

**Describe the results you expected:**
Expect to be able to start/scale services an unlimited number of times without failure

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Docker version 1.13.0, build 49bf474
```

**Output of `docker info`:**

```
Containers: 9
 Running: 3
 Paused: 0
 Stopped: 6
Images: 10
Server Version: 1.13.0
Storage Driver: devicemapper
 Pool Name: docker-8:17-2310156-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 3.537 GB
 Data Space Total: 107.4 GB
 Data Space Available: 40.73 GB
 Metadata Space Used: 5.276 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.142 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /data00/docker/lib/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /data00/docker/lib/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: 5p8zejdrwo1mrhd9te8wyz4jv
 Is Manager: true
 ClusterID: cukri7h3qcto9i6mp9cgh18n5
 Managers: 1
 Nodes: 5
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.10.11.100
 Manager Addresses:
  10.10.11.100:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 2f7393a47307a16f8cee44a37b262e8b81021e3e
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.1.12-37.6.3.el7uek.x86_64
Operating System: Oracle Linux Server 7.3
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 15.42 GiB
Name: vmlx-wd01.ibaset.com
ID: MHSZ:WK4W:CZIO:UPOU:QMQ2:DAXZ:VDFL:XAT4:E73Q:MINL:KGZB:WA5Z
Docker Root Dir: /data00/docker/lib
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-ip6tables is disabled
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
VM (VMware)

**Comment by sirlp2** [comment]
**Update - Found the cause**

Because of issue #25526 we create our service with `--publish mode=host,target=443,published=443`.  If you then increase the replicas for this service to > 1 it will fill up the `docker_gwbridge` network with stale entries.

**Comment by aboch** [comment]
Same root cause of #30938

Fixed in swarmkit by https://github.com/docker/swarmkit/pull/1952

Fix already ported to docker 1.13.x branch.

This issue will be closed when latest swarmkit is vendored to docker master.
